app/App.tsx

```diff
@@ -48,30 +50,61 @@ const App: FC = () => {
     };
   }, []);
 
-  const [incrementTimer, setIncrementTimer] = useState<NodeJS.Timer>();
-  const [decrementTimer, setDecrementTimer] = useState<NodeJS.Timer>();
+  const [fetchUpdatesTimer, setFetchUpdatesTimer] = useState<NodeJS.Timer>();
+
+  const { selectedAccount } = store.getState();
+
+  const accountId = selectedAccount?.account?.accountId ?? '';
+
+  const fetchBalance = async () => {
+    const { balance: balanceStatus } = await fullServiceApi.getBalanceForAccount({ accountId });
+    const { walletStatus } = await fullServiceApi.getWalletStatus();
+    store.dispatch(updateStatusAction(selectedAccount.account, balanceStatus, walletStatus));
+  };
+
+  const fetchLogs = async () => {
+    const transactionLogs = await fetchAllTransactionLogsForAccount(accountId);
```

Comment:
this is a different pattern than I'm used to for async redux actions. Usually I thought you would dispatch an action that hits a middleware that handles the async operations, then dispatches an update to the store. Example: [Thunks](https://redux.js.org/usage/writing-logic-thunks). Not saying we need to implement this here and now, but something to keep in mind

Reply:
Thunk didn't make sense as an implementation path when using the Connected Components style of store injection, but with the hooks style it looks a like it makes a lot more sense to do things this way. I'll do some diving in to see what that might look like!